lectures/python_advanced_features.md

```diff
@@ -290,6 +290,227 @@ tags: [raises-exception]
 max(y)
 ```
 
+## `*` and `**` Operators
+
+`*` and `**` are convenient and widely used tools to unpack lists and tuples and to allow users to define functions that use arbitrarily many arguments as input.
```

Comment:
TBH - I almost never use these operators in my code as I find them one aspect of `python` syntax that is not memorable :-). I think @jstac uses these operators when passing parameter sets between models as a use case in the economics / research analysis domain. Can we think of some possible examples that do some things in this context?

Reply:
Hi @mmcky, many thanks for your feedback. I think you are absolutely right about the use cases. I proposed writing something about this because it is also widely used in AI/ML packages and research papers (given we also have readers who are interested in data science :) ). I think my original intention is also to introduce these concepts to readers so that they can recognise and understand them; especially when they read the documentation and source code of scientific packages later, but it does feel like I have overdone some of the sections.

I think I will reorganise this section and include small functions in `qe` package and explain what they are used for in the context of the function, making it more practical, and reorienting the focus to understanding them instead of writing them. 

Please kindly let me know if this sounds better to you.